### PR TITLE
Added Even/Odd in JavaScript

### LINKED ISSUE #1392: Add Even/Odd in JavaScript
See requirements: https://sample-programs.therenegadecoder.com/projects/even-odd/
Add snippet here: https://github.com/TheRenegadeCoder/sample-programs/tree/master/archive/j/javascript

### PR DESCRIPTION
**Pull request by gmananan**
- [x] I fixed #1392 

### Code Snippets

- [x] I named the pull request using `Added/Updated <Sample Program> in <Language>` format
- [x] I created/updated the language README
  - [x] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut, developer, typing, etc.) //None added
  - [ ] I added reference link(s) to the README //No article, so not added
  - [ ] I added solution citations when necessary (see [plagiarism][2]) //Not needed

## Notes

Added even-odd in JS, and fixed one capitalization typo in the readme.

### CODE REVIEW

File: archive/j/javascript/even-odd.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const input = Number(process.argv[2]); //coerce the input into a number if possible
+if(!Number.isInteger(input)){ //if there is no input, input = undefined and the statement still prints
+    console.log('Usage: please input an integer as the first command line argument');
```
Comment (jrg94) [inline archive/j/javascript/even-odd.js]: Looks like this is a bit more detailed than we have in testing. 
```suggestion
    console.log('Usage: please input a number');
```

File: archive/j/javascript/even-odd.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+} else {
+    console.log(input%2 === 0 ? 'even' : 'odd');
```
Comment (jrg94) [inline archive/j/javascript/even-odd.js]: Actually, even and odd should be capitalized as well. 
```suggestion
    console.log(input%2 === 0 ? 'Even' : 'Odd');
```

**Comment by gmananan** [comment]
I wasn't scrupulous enough about enacting the requirements - I've incorporated the suggested changes.

File: archive/j/javascript/even-odd.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+const input = Number(process.argv[2]); //coerce the input into a number if possible
+if(!Number.isInteger(input)){ //if there is no input, input = undefined and the statement still prints
```
Comment (jrg94) [inline archive/j/javascript/even-odd.js]: Looks like this condition misses the empty input condition ("").

Comment (gmananan) [reply archive/j/javascript/even-odd.js]: You're right. I have added a conditional statement to catch that. I've tested that it catches null, undefined, floating point numbers, and strings with characters in them.

**Comment by gmananan** [comment]
If adding this extra conditional from the last review makes the code look too messy, and regex is allowed, the file could look like this instead:
`const errorMsg = 'Usage: please input a number';`
`const evenOrOdd = input => Number.parseInt(input) %2 === 0 ? 'Even' : 'Odd';`
`console.log(/^\d+$/.test(process.argv[2]) ? evenOrOdd(process.argv[2]) : errorMsg); `

JavaScript's `Number()`, `Number.parseInt()`, and `Number.parseFloat()` do not do exactly what is required in the problem statement: `Number()` coerces empty strings to 0 and keeps decimals, `Number.parseInt()` rounds decimals, and `Number.parseFloat()` obviously keeps decimals. Using any of these on the input requires performing additional tests to account for their behavior, but only one regex is required in order to test for a valid input.

**Pull request merged**